cli/ballerina-cli/src/main/java/io/ballerina/cli/task/CreateTestExecutableTask.java

```diff
@@ -0,0 +1,219 @@
+                        if (entry.getName().endsWith(ProjectConstants.JAVA_CLASS_EXT)) {
+                            excludingClassPaths.add(entry.getName().replace(File.separator, ProjectConstants.DOT)
+                                    .replace(ProjectConstants.JAVA_CLASS_EXT, ""));
+                        }
+                    });
+                    zipFile.close();
+                }
+
+                // Create the single fat jar for all the test modules that includes the test suite json
+                EmitResult result = jBallerinaBackend.emit(
+                        JBallerinaBackend.OutputType.TEST,
+                        testExecutablePath,
+                        testExecDependencies,
+                        TestUtils.getJsonFilePath(testCachePath),
+                        TestUtils.getJsonFilePathInFatJar(File.separator),
+                        excludingClassPaths,
+                        ProjectConstants.EXCLUDING_CLASSES_FILE
+                );
+                emitDiagnostics.addAll(result.diagnostics().diagnostics());
+            }
+
+            if (project.buildOptions().dumpBuildTime()) {
+                BuildTime.getInstance().emitArtifactDuration = System.currentTimeMillis() - start;
+                BuildTime.getInstance().compile = false;
+            }
+
+            // Print warnings for conflicted jars
+            if (!jBallerinaBackend.conflictedJars().isEmpty()) {
+                out.println("\twarning: Detected conflicting jar files:");
+                for (JBallerinaBackend.JarConflict conflict : jBallerinaBackend.conflictedJars()) {
+                    out.println(conflict.getWarning(project.buildOptions().listConflictedClasses()));
+                }
+            }
+
+            if (!emitDiagnostics.isEmpty()) {
+                emitDiagnostics.forEach(d -> out.println("\n" + d.toString()));
+            }
+        } catch (ProjectException | IOException e) {
+            throw createLauncherException(e.getMessage());
+        }
+        // notify plugin
+        // todo following call has to be refactored after introducing new plugin architecture
+        // Similar case as in CreateExecutableTask.java
+        notifyPlugins(project, target);
+    }
+
+    private boolean createTestSuiteForCloudArtifacts(Project project, JBallerinaBackend jBallerinaBackend,
+                                                     Target target, Map<String, TestSuite> testSuiteMap) {
+        boolean report = project.buildOptions().testReport();
+        boolean coverage = project.buildOptions().codeCoverage();
+
+        TestProcessor testProcessor = new TestProcessor(jBallerinaBackend.jarResolver());
+        List<String> moduleNamesList = new ArrayList<>();
+        List<String> updatedSingleExecTests;
+        List<String> mockClassNames = new ArrayList<>();
+
+        boolean status = RunTestsTask.createTestSuitesForProject(project, target, testProcessor, testSuiteMap,
```

Comment:
What does this `status` mean? Shall we have a better name that reflects the value it hold?

Reply:
Status indicates whether the test suite has been created if there are any tests. The value is false if no tests found. And yes, I Agree. I have done the relevant changes in 89c50f2 commit.